### ISSUE TITLE
Fragment factories

### ISSUE THREAD

**Issue by egor-n** [eugenkiss/chanobol#67]
Hey, I've been following this project for quite some time and I must say you're doing great work on documenting every step you make. Your comments and the links you provide when fixing bugs are very handy.

I stumbled across [this](https://github.com/eugenkiss/chanobol/blob/master/src/main/java/anabolicandroids/chanobol/ui/threads/ThreadsFragment.java#L50) factory method and I'm pretty sure that's not how it should be done. The framework will recreate the fragment, but it won't restore member fields' values. You should be using a `Bundle` and setting it as fragment's arguments like so:

```java
public static ThreadsFragment create(String name) {
    ThreadsFragment f = new ThreadsFragment();
    Bundle args = new Bundle();
    args.putString("name", name);
    f.setArguments(args);
    return f;
}
```

And then use this value like so:

```java
public void onActivityCreated(@Nullable Bundle savedInstanceState) {
    name = getArguments().getString("name");
    // ...
}
```

The main point is that when Android framework recreates the fragment, it will restore its arguments as well. [This answer](http://stackoverflow.com/a/9245510/1348215) can provide some more information.

**Comment by eugenkiss** [comment]
Hey, thanks for the compliment and your suggestion/reference! You are the first contributor, that makes my happy :smile:.

You are right, I'm not doing it correctly at the moment :sweat:. I've been postponing the important issue of activity/fragment state storing and recreation (for example on configuration changes like switching from portrait to landscape layout) for quite some time. And I'm pretty sure at least one of the bugs I encountered stems directly from *not* recreating the state of a fragment correctly. So once I've finished all the stuff I planned to get done for 1.0 I'm going to tackle this issue.
